### PR TITLE
Crash reporter

### LINKED ISSUE #6: Crash webhook

### PR DESCRIPTION
**Pull request by white-axe**
**Connections**
* Closes #6

**Description**
Luminol now restarts after panicking on both native and web platforms. After restarting, it shows a window asking the user if they want to send the crash report to the developers. The crash report is a JSON object sent as a POST request; it contains the panic message as well as the Git revision, target triple and profile (debug/release).

It reports to http://localhost:3246 right now for easy testing. This should be changed in crates/ui/src/windows/reporter.rs to a real URL at some point. Don't worry about CORS - web builds can send crash reports even if the endpoint doesn't set CORS headers.

Restarting after panics is done on Unix-like systems by using one of the exec system calls to replace the current process without creating a new one. On Windows, it spawns a background process and then terminates the original one. For WebAssembly, it's done by reloading the WebAssembly modules. Reports are persisted across restarts in native builds by creating a temporary file and in web builds by saving the report as a JavaScript global variable.

I was originally going to try using `std::panic::catch_unwind` to catch panics, but this doesn't work in WebAssembly due to the panic_abort panic handler and it causes segfaults and other signs of undefined behaviour in native builds somewhere inside of winit.

There are also some related improvements to web builds:

- Luminol now completely caches itself after loading for the first time so that the browser doesn't have to download the WebAssembly module again every time a panic occurs. No internet access is required after the first page load.
- The Git revision is now determined from the Trunk hooks instead of with the git-version macro so that it can correctly determine if the code has uncommitted changes. It should now append or not append "-modified" to the Git revision properly instead of always appending "-modified" regardless of whether there are uncommitted changes.
- Event handlers for mouse and keyboard and the like are now unregistered after panics for performance reasons.
- Clippy no longer errors out when checking luminol-web, so we can now see clippy lints for crates that depend on luminol-web.

**Testing**
Right now, a panic can be triggered by attempting to create or edit events in the map editor.

Testing for reception of reports on localhost can be done with a simple echo server like this one: https://github.com/cdfuller/echo-server. If you use this one in particular, make sure to run it with the `-v` flag so that it prints the contents of requests. Whatever echo server is used, test that it is able to receive reports from both native and web builds of Luminol.

It would also be a good idea to check that a real, internet-facing server is able to receive reports, especially that one that does not support CORS is able to receive reports from a web build of Luminol.

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
- [x] Run `cargo build --release` 
- [x] If applicable, run `trunk build --release`

### CODE REVIEW

File: crates/core/Cargo.toml
Comment (somedevfox) [inline crates/core/Cargo.toml:1]: Changing git-version from pulling a version from the workspace to a specific version can result in incompatibility between the core and other crates in the future.

**Review by somedevfox — CHANGES_REQUESTED** [review]

File: crates/core/Cargo.toml
Comment (white-axe) [reply crates/core/Cargo.toml:1]: I didn't change how the git version is determined in native builds. In web builds, I only changed it to determine the git version inside of hooks/trunk_enable_build_std_pre.sh because the git-version macro incorrectly says that the code has been modified relative to the latest commit even when it hasn't been. So the version is still automatically determined.

**Comment by white-axe** [comment]
Since I'd already changed how the Git version is calculated in web builds, I've also moved the Git version into the luminol crate (#89). This will prevent the version from being different in different crates if the crates are not all at the same revision.

**Review by melody-rs — APPROVED** [review]

**Comment by white-axe** [comment]
I can't merge this until @somedevfox approves

**Review by melody-rs — APPROVED** [review]

**Review by somedevfox — APPROVED** [review]

**Pull request merged**
